stock_picking_batch_extended_account/models/stock_batch_picking.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2020 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class StockBatchPicking(models.Model):
+    _inherit = 'stock.picking.batch'
+
+    def action_print_invoices(self):
+        invoices = self.mapped('picking_ids.sale_id.invoice_ids')
+        if not invoices:
+            raise UserError(_('Nothing to print.'))
+        report = self.env.ref('account.account_invoices_without_payment')
+        return report.report_action(invoices)
+
+    def _validate_and_invoice(self, func_name):
+        picking_to_invoice = self.mapped("picking_ids").filtered(
+            lambda r: r.partner_id.batch_picking_auto_invoice)
+        self_with_ctx = self.with_context(
+            picking_to_invoice_in_batch=picking_to_invoice.ids)
+        return getattr(super(StockBatchPicking, self_with_ctx), func_name)()
```

Comment:
Why doing this in such weird form instead of simply calling this method without the super, and calling the super on each method? You only save one line and this gets less readable.